.github/scripts/update_schutzfile.py

```diff
@@ -0,0 +1,38 @@
+#!/bin/python3
+
+import json
+import os
+import re
+
+repo_files = os.listdir("/repo/")
+singletons = []
+# Get a list of all repositories that contain 'singleton'
+for repo_file in repo_files:
+    with open(os.path.join("repo", repo_file), "r") as f:
+        data = json.load(f)
+    if "singleton" in data.keys():
+        singletons.append(data["snapshot-id"])
+
+with open(os.path.join(os.environ.get("REPO"), "Schutzfile"), "r") as f:
+    data = json.load(f)
+
+urls = []
+arches = ["x86_64", "aarch64"]
+# Replace snapshot SUFFIX in repositories that don't have 'singleton' present
+for key in data.keys():
+    if "repos" in data[key].keys():
+        for i, _ in enumerate(data[key]["repos"]):
+            for arch in arches:
```

Comment:
If the previous suggestion is applied, we can make this `for arch_repos in repo` and just use `arch_repo` onwards, e.g.:

```
if any(singleton in arch_repos[0]["baseurl"] for singleton in singletons):
```

Reply:
I got a bit stuck on this cause `repo` in this case is still a dict which has a string `file` and then two lists for each arch, but was able to do it without the enumeration in the end.